app/controllers/spree/products_controller_decorator.rb

```diff
@@ -1,9 +1,18 @@
 module Spree
   ProductsController.class_eval do
-    after_action :recently_viewed, only: :show
+    skip_before_action :set_current_order, only: :recently_viewed
+    after_action :save_recently_viewed, only: :recently_viewed
```

Comment:
@damianlegawiec I think this callback should be after `#show` and not `#recently_viewed`.